tests/Cms/Api/ApiCollectionTestCase.php

```diff
@@ -10,22 +10,37 @@ class ApiCollectionTestCase extends TestCase
 {
 	protected $api;
 	protected $app;
-	protected $tmp = __DIR__ . '/tmp';
 
 	public function setUp(): void
 	{
+		if ($this->hasTmp() === true) {
+			Dir::remove(static::TMP);
+		}
+
 		$this->app = new App([
 			'roots' => [
-				'index' => $this->tmp,
+				'index' => $this->hasTmp() ? static::TMP : '/dev/null',
 			],
 		]);
 
 		$this->api = $this->app->api();
-		Dir::make($this->tmp);
 	}
 
 	public function tearDown(): void
 	{
-		Dir::remove($this->tmp);
+		App::destroy();
+
+		if ($this->hasTmp() === true) {
+			Dir::remove(static::TMP);
+		}
+	}
+
+	/**
+	 * Checks if the test class extending this test case class
+	 * has defined a temporary directory
+	 */
+	protected function hasTmp(): bool
+	{
+		return defined(get_class($this) . '::TMP');
```

Comment:
Maybe good to refactor this and some other parts to have one base TestCase class for all our tests. But doesn't need to be within this PR.